src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,37 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        StringBuilder report = new StringBuilder("Отчёт за период ");
+        LocalDate dateF = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate dateT = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        if (dateF.isAfter(dateT)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        report.append(dateFrom).append(" - ").append(dateTo).append("\n");
+
+        for (String name : names) {
+            int salary = 0;
+            for (String str : data) {
+                if (str.contains(name)) {
+                    String[] arrStr = str.split(" ");
+                    if (LocalDate.parse(arrStr[0], DATE_TIME_FORMATTER)
+                            .compareTo(dateF) >= 0
+                            && LocalDate.parse(arrStr[0], DATE_TIME_FORMATTER)
+                            .compareTo(dateT) <= 0) {
```

Comment:
try to use isAfter() or isBefore() those are more relevant and you'll get rid of comparing with zero every time

Reply:
@dmytro-hryhoruk  Why should we use isAfter() and isBefore()? In test we have situation where day of work equals with dateTo, and in this case we need to use compareTo() or increment dateTo in start.